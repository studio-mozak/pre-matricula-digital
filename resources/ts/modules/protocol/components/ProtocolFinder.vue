<template>
  <div>
    <div class="row">
      <protocol-pre-find />
    </div>
    <div class="d-flex justify-content-md-center mt-5 row">
      <div class="col-md-3">
        <x-btn
          data-test="btn-back"
          color="gray-500"
          outline
          class="w-100"
          label="Voltar"
          no-caps
          no-wrap
          @click="prev"
        />
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import ProtocolPreFind from '@/modules/protocol/components/ProtocolPreFind.vue';
import XBtn from '@/components/elements/buttons/XBtn.vue';
import { useRouter } from 'vue-router';

const router = useRouter();

const prev = () => {
  router.push('/');
};
</script>
