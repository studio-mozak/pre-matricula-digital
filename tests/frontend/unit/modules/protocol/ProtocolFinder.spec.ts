import { VueWrapper, mount } from '@vue/test-utils';
import ProtocolFinder from '@/modules/protocol/components/ProtocolFinder.vue';
import { expect, vi } from 'vitest';

vi.mock('vue-router/dist/vue-router.mjs', () => ({
  useRouter: () => ({
    push: spyPush,
  }),
}));

const spyPush = vi.fn();

const onMounted = async () => {
  const wrapper = mount(ProtocolFinder, {
    global: {
      plugins: [],
      stubs: [],
    },
  });

  await wrapper.vm.$nextTick();

  return wrapper;
};

describe('ProtocolFinder', () => {
  let wrapper: VueWrapper<InstanceType<typeof ProtocolFinder>>;

  beforeEach(async () => {
    wrapper = await onMounted();

    await wrapper.vm.$nextTick();
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  test('the component exists', () => {
    expect(wrapper).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('check if protocol-pre-find components exists', () => {
    const component = wrapper.findComponent({
      name: 'protocol-pre-find',
    });

    expect(component).toBeTruthy();
  });

  test('when I click in "go back" button, it should go to "/"', async () => {
    const goBackButton = wrapper.find('[data-test="btn-back"]');

    goBackButton.trigger('click');

    await wrapper.vm.$nextTick();

    expect(spyPush).toHaveBeenCalledTimes(1);
    expect(spyPush).toHaveBeenCalledWith('/');
  });
});
