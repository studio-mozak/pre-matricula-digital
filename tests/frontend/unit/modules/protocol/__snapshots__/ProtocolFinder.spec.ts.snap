// Vitest Snapshot v1

exports[`ProtocolFinder > should match snapshot 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-md-8 mt-5 offset-md-2"
    >
      <h2
        class="font-muli-20-primary mb-4"
      >
         Identifique o(a) aluno(a) para encontrar o(s) protocolo(s): 
      </h2>
      <p
        class="text-justify font-hind"
      >
         Selecione uma das opções abaixo para validar se o(a) 
        <strong>
          aluno(a)
        </strong>
         já possui alguma inscrição de pré-matrícula. Caso não seja possível identificá-lo com a opção selecionada, você pode escolher outro campo para validar a existência de protocolo(s). 
      </p>
      <div
        class="mt-5"
      >
        <button
          class="btn btn-block btn-outline-primary mb-2"
          data-test="btn-1"
          type="button"
        >
          <span
            class="text-uppercase font-weight-lighter font-hind"
            style="font-size: 16px;"
          >
             CPF do(a) aluno(a) 
          </span>
        </button>
        <button
          class="btn btn-block btn-outline-primary mb-2"
          data-test="btn-2"
          type="button"
        >
          <span
            class="text-uppercase font-weight-lighter font-hind"
            style="font-size: 16px;"
          >
             RG do(a) aluno(a) 
          </span>
        </button>
        <button
          class="btn btn-block btn-outline-primary mb-2"
          data-test="btn-3"
          type="button"
        >
          <span
            class="text-uppercase font-weight-lighter font-hind"
            style="font-size: 16px;"
          >
             Certidão de nascimento do(a) aluno(a) 
          </span>
        </button>
        <button
          class="btn btn-block btn-outline-primary mb-2"
          data-test="btn-4"
          type="button"
        >
          <span
            class="text-uppercase font-weight-lighter font-hind"
            style="font-size: 16px;"
          >
             Nome e data de nascimento do(a) aluno(a) 
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="d-flex justify-content-md-center mt-5 row"
  >
    <div
      class="col-md-3"
    >
      <button
        class="x-btn x-btn-item non-selectable no-outline x-btn--outline x-btn--rectangle text-gray-500 border-gray-500 x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase w-100"
        data-test="btn-back"
        role="button"
        tabindex="0"
        type="button"
      >
        <span
          class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
        >
          <span
            class="block"
          >
            Voltar
          </span>
        </span>
        <transition-stub
          appear="false"
          css="true"
          name="x-transition--fade"
          persisted="false"
        >
          <!---->
        </transition-stub>
      </button>
    </div>
  </div>
</div>
`;
